tests/cancun/eip1153_tstore/test_tstorage_execution_contexts.py

```diff
@@ -0,0 +1,208 @@
+        ),
+        "callee_bytecode": Op.TSTORE(0),  # calling tstore fails
+        "expected_caller_storage": {0: 0, 1: 420},
+        "expected_callee_storage": {},
+    }
+    STATICCALL_CAN_CALL_TLOAD = {
+        # TODO: Not a very useful test; consider removing after implementing ethereum/tests
+        # staticcall tests
+        "pytest_param": pytest.param(id="staticcalled_context_can_call_tload"),
+        "description": ("TSTORE0003: A STATICCALL callee can not use transient storage."),
+        "caller_bytecode": (
+            Op.TSTORE(0, 420)
+            + Op.SSTORE(0, Op.STATICCALL(Op.GAS(), callee_address, 0, 0, 0, 0))
+            + Op.SSTORE(1, Op.TLOAD(0))
+        ),
+        "callee_bytecode": Op.TLOAD(0),  # calling tload does fail the call
+        "expected_caller_storage": {0: 1, 1: 420},
+        "expected_callee_storage": {},
+    }
+    CALLCODE = {
+        "description": (
+            "TSTORE0004: Caller and callee contracts share transient storage "
+            "when callee is called via CALLCODE."
+        ),
+        "caller_bytecode": (
+            Op.TSTORE(0, 420)
+            + Op.SSTORE(0, Op.CALLCODE(Op.GAS(), callee_address, 0, 0, 0, 0, 0))
+            + Op.SSTORE(1, Op.TLOAD(0))
+            + Op.SSTORE(4, Op.TLOAD(1))
+        ),
+        "callee_bytecode": (
+            Op.SSTORE(2, Op.TLOAD(0)) + Op.TSTORE(1, 69) + Op.SSTORE(3, Op.TLOAD(1))
+        ),
+        "expected_caller_storage": {0: 1, 1: 420, 2: 420, 3: 69, 4: 69},
+        "expected_callee_storage": {},
+    }
+    CALLCODE_WITH_REVERT = {
+        "description": (
+            "TSTORE0005: Caller and callee contracts share transient storage "
+            "when callee is called via CALLCODE. Transient storage usage "
+            "from sub-call upon revert."
+        ),
+        "caller_bytecode": (
+            Op.TSTORE(0, 420)
+            + Op.TSTORE(1, 420)
+            + Op.SSTORE(0, Op.CALLCODE(Op.GAS(), callee_address, 0, 0, 0, 0, 0))
+            + Op.SSTORE(1, Op.TLOAD(0))
+            + Op.SSTORE(2, Op.TLOAD(1))
+        ),
+        "callee_bytecode": Op.TSTORE(1, 69) + Op.REVERT(0, 0),
+        "expected_caller_storage": {0: 0, 1: 420, 2: 420},
+        "expected_callee_storage": {},
+    }
+    DELEGATECALL = {
+        "description": (
+            "TSTORE0006: Caller and callee contracts share transient storage "
+            "when callee is called via DELEGATECALL."
+        ),
+        "caller_bytecode": (
+            Op.TSTORE(0, 420)
+            + Op.SSTORE(0, Op.DELEGATECALL(Op.GAS(), callee_address, 0, 0, 0, 0))
+            + Op.SSTORE(1, Op.TLOAD(0))
+            + Op.SSTORE(4, Op.TLOAD(1))
+        ),
+        "callee_bytecode": (
+            Op.SSTORE(2, Op.TLOAD(0)) + Op.TSTORE(1, 69) + Op.SSTORE(3, Op.TLOAD(1))
+        ),
+        "expected_caller_storage": {0: 1, 1: 420, 2: 420, 3: 69, 4: 69},
+        "expected_callee_storage": {},
+    }
+    DELEGATECALL_WITH_REVERT = {
```

Comment:
It might be worth it to add `DELEGATECALL_WITH_INVALID`, and `CALLCODE_WITH_INVALID`, which would be basically the same test case but with a `Op.INVALID` at the end instead of revert, and using much less gas on the call since running into an invalid opcode is similar to an OOG error.

Reply:
Added in e7cb2148.